packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -826,13 +827,14 @@ class _InkResponseState extends State<_InkResponseStateWidget>
     void handleInkRemoval() {
       assert(_highlights[type] != null);
       _highlights[type] = null;
-      updateKeepAlive();
+      if (active)
+        updateKeepAlive();
```

Comment:
is it possible that we'll be reactivated after this point?

Reply:
The error is caused by `highlightsExist` is true, which causes `wantKeepAlive` to be true.

This is caused by calling `Highlight.dispose` after `deactivate`.

Two ways have been found to reproduce this, call `Highlight.dispose` when `State.dispose`, if `_highlights.values` is more than one then `highlightsExist` is true.

The second is that `State.deactivate` is followed by callback to `_handleTapCancel`.


For the first one, we can set all `_highlights.values` to null and then execute `highlight.dispose` to solve the problem.

I don't have a good solution for the second one, so I'm using `State.active` to solve it.

And I think that exposing `State.active` will help to solve similar requirements or issues in the future.

Reply:
Sorry I feel like I'm missing some critical information here! I was just wondering if we need to call updateKeepAlive in reactivate() since we don't call it when handleInkRemoval is called after deactivate. I'm not sure what error you're referring to in your response.

Reply:
Sorry, I misunderstood what you meant.

You are right, we should call `updateKeepAlive` in `reactivate`, because wantKeepAlive's value maybe changed.

But it should be better if we call `updateKeepAlive` in `AutomaticKeepAliveClientMixin. reactivate`?

Thanks for your guidance.

Reply:
done, after reading the `AutomaticKeepAlive` code I realized that I shouldn't call `updateKeepAlive` in `AutomaticKeepAliveClientMixin.reactivate`.